Lemnisk latest doc (#1136)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/lemnisk/lemnisk-cloud-mode.mdx

---
title: "Lemnisk Cloud Mode"
description: Send events to Lemnisk using RudderStack cloud mode.
---

RudderStack lets you send your event data to Lemnisk via the <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link>. Find the open source code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/lemnisk">GitHub repository</a>.

RudderStack supports the following Lemnisk platforms:

## DIAPI

> When you choose to send events to Lemnisk via the server-side cloud mode, Lemnisk' [HTTP API](https://lemnisk.gitbook.io/lemnisk-platform-documentation/sources/http-api) is used to send your website or app data.

Reviewer: ```suggestion
When you choose to send events to Lemnisk via the server-side cloud mode, RudderStack uses Lemnisk's [HTTP API](https://lemnisk.gitbook.io/lemnisk-platform-documentation/sources/http-api) to send your website or app data.
```

### Track

The <Link to="/event-spec/standard-events/track/">`track`</Link> call lets you record the user actions along with any properties associated with them.

A sample `track` call is shown below:

```javascript
rudderanalytics.track("Product Reviewed", {
  review_id: "12345",
  product_id: "123",
  rating: 3.0,
  review_body: "Average product, expected much more."
})
```

> ### Property mappings

Reviewer: Should be H4 as it's associated with `track`, IMO

The following table lists the mappings between the RudderStack and Lemnisk properties:

| RudderStack property   | Lemnisk property  | Data type |
| :-------------| :-----------------| :-----------------| 
|  `userId` <br/> `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `userId` | String |
| `properties` | `properties` | Object |
| `event`  | `eventname` | String |
| `config.srcid` | `srcId` | String |
| `config.diapiWriteKey`  | `writekey` | String |

> 

Reviewer: ```suggestion
```

## Pixel Listener

> When you choose to send events to Lemnisk via the web cloud mode, Lemnisk' [JavaScript SDK](https://lemnisk.gitbook.io/lemnisk-platform-documentation/sources/javascript-sdk) is used to send data from your website.

Reviewer: ```suggestion
When you choose to send events to Lemnisk via the web cloud mode, RudderStack uses Lemnisk's [JavaScript SDK](https://lemnisk.gitbook.io/lemnisk-platform-documentation/sources/javascript-sdk) to send data from your website.
```

> ## Identify

Reviewer: ```suggestion
### Identify
```

Reviewer: Is this call supported only for Pixel Listener?

You can use the <Link to="/event-spec/standard-events/identify">`identify`</Link> call to identify a user and associate them to their actions. It also lets you record any traits about them like their name, email, etc.

A sample `identify` call is shown below:

```javascript
rudderanalytics.identify("hashed_user_id", {
  name: "Alex Keener",
  email: "alex@example.com",
})
```

> ### Property mappings

Reviewer: Should be H4 if that's the case.

The following table lists the mappings between the RudderStack and Lemnisk properties:

| RudderStack property   | Lemnisk property  |  Data type |
| :-------------| :-----------------| :-----------------| 
|  `userId` <br/> `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `userId` <br/> `id` | String |
| `context` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `context` | Object |
| `traits` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>| `customerProperties` | Object |
| `messageId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>| `messageId` | String |

> ## Track

Reviewer: ```suggestion
### Track
```

Reviewer: Same as `identify`.

A sample `track` call is shown below:

```javascript
rudderanalytics.track("Product Reviewed", {
  review_id: "12345",
  product_id: "123",
  rating: 3.0,
  review_body: "Average product, expected much more."
})
```

### Property mappings

The following table lists the mappings between the RudderStack and Lemnisk properties:

| RudderStack property   | Lemnisk property  |  Data type |
| :-------------| :-----------------| :-----------------| 
|  `userId` <br/> `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `userId` <br/> `id` | String |
| `event` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `event` | String |
| `properties`   | `properties` | Object |
| `timestamp` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>| `originalTimestamp` | String |
| `messageId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>| `messageId` | String |
| `context` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `context` | Object |
| `properties`   | `properties` | Object |

> ## Page

Reviewer: same as above.

RudderStack supports the following Lemnisk platforms:

## DIAPI

When you choose to send events to Lemnisk via the server-side cloud mode, Lemnisk' [HTTP API](https://lemnisk.gitbook.io/lemnisk-platform-documentation/sources/http-api) is used to send your website or app data.

### Track

The <Link to="/event-spec/standard-events/track/">`track`</Link> call lets you record the user actions along with any properties associated with them.

A sample `track` call is shown below:

```javascript
rudderanalytics.track("Product Reviewed", {
  review_id: "12345",
  product_id: "123",
  rating: 3.0,
  review_body: "Average product, expected much more."
})
```

### Property mappings

The following table lists the mappings between the RudderStack and Lemnisk properties:

| RudderStack property   | Lemnisk property  | Data type |
| :-------------| :-----------------| :-----------------| 
|  `userId` <br/> `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `userId` | String |

> |  `email` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `email` | String |

Reviewer 2: We can remove email mapping as we released a fix for this afterwards

| `properties` | `properties` | Object |
| `event`  | `eventname` | String |

> | `track` | `type` | String |

Reviewer 2: We can remove type property as well.

| `event`  | `eventname` | String |
| `config.srcid` | `srcId` | String |
| `config.diapiWriteKey`  | `writekey` | String |

## Pixel Listener

When you choose to send events to Lemnisk via the web cloud mode, Lemnisk' [JavaScript SDK](https://lemnisk.gitbook.io/lemnisk-platform-documentation/sources/javascript-sdk) is used to send data from your website.

## Identify

You can use the <Link to="/event-spec/standard-events/identify">`identify`</Link> call to identify a user and associate them to their actions. It also lets you record any traits about them like their name, email, etc.

A sample `identify` call is shown below:

```javascript
rudderanalytics.identify("hashed_user_id", {
  name: "Alex Keener",
  email: "alex@example.com",
})
```

### Property mappings

The following table lists the mappings between the RudderStack and Lemnisk properties:

| RudderStack property   | Lemnisk property  |  Data type |
| :-------------| :-----------------| :-----------------| 
|  `userId` <br/> `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `userId` <br/> `id` | String |

> | `type` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>| `type` | String |

Reviewer 2: Same as above

| `properties`   | `properties` | Object |
| `timestamp` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>| `originalTimestamp` | String |
| `messageId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>| `messageId` | String |
| `context` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `context` | Object |
| `properties`   | `properties` | Object |

## Page

The <Link to="/event-spec/standard-events/page/">`page`</Link> call lets you record when a user views a page of your website, along with any optional properties about the page.

A sample `page` call is shown below:

```javascript
rudderanalytics.page("Home")
```

### Property mappings

The following table lists the mappings between the RudderStack and Lemnisk properties:

| RudderStack property   | Lemnisk property  | Data type |
| :-------------| :-----------------| :-----------------| 
|  `userId` <br/> `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `userId` <br/> `id` | String |
| `name` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `name` | String |

> | `type` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `type` | String |

Reviewer 2: Same as above
